Submitted By:            Randy McMurchy <randy_at_linuxfromscratch_dot_org>
Date:                    2005-10-04
Initial Package Version: 0.7.11
Upstream Status:         N/A (package is no longer maintained)
Origin:                  Randy McMurchy
Description:             Fixes GCC-4.0.x build problems

$LastChangedBy$
$Date$


diff -Naur soup-0.7.11-orig/src/libsoup/soup-queue.c soup-0.7.11/src/libsoup/soup-queue.c
--- soup-0.7.11-orig/src/libsoup/soup-queue.c	2002-12-02 16:02:23.000000000 +0000
+++ soup-0.7.11/src/libsoup/soup-queue.c	2005-10-04 18:30:50.000000000 +0000
@@ -30,7 +30,7 @@
 #include "soup-ssl.h"
 #include "soup-transfer.h"
 
-static GSList *soup_active_requests = NULL, *soup_active_request_next = NULL;
+GSList *soup_active_requests = NULL, *soup_active_request_next = NULL;
 
 static guint soup_queue_idle_tag = 0;
 
diff -Naur soup-0.7.11-orig/src/libsoup/soup-server.c soup-0.7.11/src/libsoup/soup-server.c
--- soup-0.7.11-orig/src/libsoup/soup-server.c	2002-07-26 15:34:12.000000000 +0000
+++ soup-0.7.11/src/libsoup/soup-server.c	2005-10-04 18:30:50.000000000 +0000
@@ -312,7 +312,10 @@
 	/*
 	 * Get request HTTP method
 	 */
-	(gchar *) msg->method = g_strdup (g_getenv ("REQUEST_METHOD"));
+	/* (gchar *) msg->method = g_strdup (g_getenv ("REQUEST_METHOD")); */
+
+	const method = g_strdup (g_getenv ("REQUEST_METHOD"));
+	(gchar *) msg->method;
 
 	/* 
 	 * Get content length of request body
